src/components/Authentication/Login/Login.jsx

```diff
@@ -0,0 +1,116 @@
+import { useFormik } from "formik";
+import { loginSchema } from "../../../../validationSchemas";
+import { useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+function Login() {
+  const navigate = useNavigate();
+  const [error, setError] = useState("");
+
+  const formik = useFormik({
+    initialValues: {
+      email: localStorage.getItem("email") || "",
+      password: localStorage.getItem("password") || "",
+    },
+    validationSchema: loginSchema,
+    onSubmit: () => {
+      const storedUserData = JSON.parse(localStorage.getItem("userData"));
+      if (
+        storedUserData &&
+        formik.values.email === storedUserData.email &&
+        formik.values.password === storedUserData.password
+      ) {
+        navigate("/product-list");
+      } else {
+        setError("Email or password are incorrect.");
+      }
+    },
```

Comment:
Firstly: 
- the onSubmit function has a parameter, lets call it `values` eg: onSubmit(values) and we can take the values from the form from there.
- we need to console log those values 
- we then need to check for the validation/errors. 
- if there are no errors we need to store a token - a random one - to the localStorage 
eg: localstorage.set("authToken", "my-test-token");
- and we need to dispatch an event that the authentication had place
eg: [you can see here how it's supposed to look](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events)

Now we have a token stored so we can bypass the login and we have sent an event for the whole app to know that we have authenticated.
There's one more thing to do, in the App.jsx ---